### ISSUE TITLE
IOS Wallet having duplicate events in subsequent VC transfers

### ISSUE THREAD

**Issue by krishnakumar4a4** [mosip/inji#546]
This is observed during internal dev testing between IOS Wallet and Android Verifier. From the surface multiple VC transfers seems to work. But we can observe duplicate events being fired which may lead to inconsistent behaviour. 

One observation we had was when after 3 or 4 VC transfers, the app became un responsive and few mins also crashed (observed only once so far). We are yet investigate attribute if this behaviour caused the crash. Instead of that, we think fixing the duplicate events is necessary and see if app is consistent with multiple VC transfers.

Wallet:
- Iphone 13 with IOS 16.2 and BLE 5.0

Verifier
- Infinix Hot 11 with Android 11 and BLE 5.1

**Comment by Alka1703** [comment]
The following PRs addresses the duplicate events being fired on the iOS wallet side. 
PR for mosip/tuvali: https://github.com/mosip/tuvali/pull/26
PR for mosip/inji:https://github.com/mosip/inji/pull/584

The PRs have been merged in 24-02-2023 Release and the latest commit id on mosip/inji/develop is `cb47493`

**Comment by shreeThaanuTw** [comment]
**Adding few more AC :** 

-  Remove Singleton and replace Notifications with Delegate.

**Comment by ravikp** [comment]
Re-opening the card since there are a few places where the use of singleton classes are observed in the iOS Tuvali repository.

**Comment by ravikp** [comment]
We'll move the work to refactor into a new card.
